### PR TITLE
MM-48215: add Dockerfile for running python scripts

### PR DESCRIPTION
**Pull request by ifoukarakis**
### Summary

- [x] Add `Dockerfile` for python utilities.
- [x] Add `.dockerignore` to exclude copying files not needed for 
- [x] Update `Makefile` to include docker targets, following example from [here](https://github.com/mattermost/rtcd/blob/master/Makefile).
- [x] Remove not needed dependency in order to simplify build (no need to install git).
- [x] Updated github action to build, push and sign docker image to [docker hub](https://hub.docker.com/repository/docker/mattermost/mattermost-data-warehouse/general).

### Dockerfile summary

The Dockerfile is a multi-stage build. It is using `python:3.x-slim` as the base image. There are three stages:
- `base` stage standardizes python environment preferences. 
- `builder` stage installs any prerequisites required to install dependencies, poetry (dependency management) and finally uses poetry to install dependencies (excluding dev dependencies) and builds a wheel package containing the python scripts.
- `final` stage copies dependencies from `builder` and installs the wheel package. 

A better approach would have been to start with a [distroless python](https://github.com/GoogleContainerTools/distroless) image. Unfortunately the distroless python is still experimental and should be avoided on production.

### CODE REVIEW

**Comment by phoinixgrr** [comment]
There are some conflicts that need resolving @ifoukarakis

**Comment by ifoukarakis** [comment]
> There are some conflicts that need resolving @ifoukarakis

Oops, missed that! The conflicts are now resolved.

**Comment by phoinixgrr** [comment]
There are some conflicts that need resolving @ifoukarakis

File: .github/workflows/ci-python.yml [hunk 1]
```diff
@@ -52,3 +54,25 @@ jobs:
...
+        run: make docker-lint
+
+      - name: Build & push docker image
+        run: make docker-login && make docker-build && make docker-push && make docker-sign && make docker-verify
+        env:
+          DOCKER_USER: ${{ secrets.DOCKER_USER }}
+          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+          COSIGN_KEY: ${{ secrets.COSIGN_KEY }}
+          COSIGN_PASSWORD: ${{ secrets.COSIGN_PASSWORD }}
+          COSIGN_PUBLIC_KEY: ${{ secrets.COSIGN_PUBLIC_KEY }}
```
Comment (phoinixgrr) [inline .github/workflows/ci-python.yml:78]: Let's add a NL here

File: .github/workflows/ci-python.yml [hunk 2]
```diff
@@ -52,3 +54,25 @@ jobs:
...
+      - name: checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+
+      - name: Lint docker image
+        run: make docker-lint
+
+      - name: Build & push docker image
+        run: make docker-login && make docker-build && make docker-push && make docker-sign && make docker-verify
```
Comment (phoinixgrr) [inline .github/workflows/ci-python.yml:72]: I'd suggest to split-out the `&& make docker-sign && make docker-verify` steps to a new dedicated step.

**Review by phoinixgrr — APPROVED** [review]
Awesome work @ifoukarakis 🎖️ 🏅 🥇 

:shipit:

**Pull request merged**